lib/mvi/service.rb

```diff
@@ -31,15 +32,46 @@ def find_profile(user)
       raw_response = perform(:post, '', create_profile_message(user), soapaction: OPERATIONS[:find_profile])
       MVI::Responses::FindProfileResponse.with_parsed_response(raw_response)
     rescue Faraday::ConnectionFailed => e
-      Rails.logger.error "MVI find_profile connection failed: #{e.message}"
+      log_message_to_sentry("MVI find_profile connection failed: #{e.message}", :error)
       MVI::Responses::FindProfileResponse.with_server_error
     rescue Common::Client::Errors::ClientError, Common::Exceptions::GatewayTimeout => e
-      Rails.logger.error "MVI find_profile error: #{e.message}"
+      log_message_to_sentry("MVI find_profile error: #{e.message}", :error)
       MVI::Responses::FindProfileResponse.with_server_error
+    rescue MVI::Errors::Base => e
+      mvi_error_handler(user, e)
     end
 
     private
 
+    # TODO: Possibly consider adding Grafana Instrumentation here too
+    def mvi_error_handler(user, e)
+      case e
+      when MVI::Errors::DuplicateRecords
+        log_message_to_sentry('MVI Duplicate Record', :warn, error_context(user))
+      when MVI::Errors::RecordNotFound
+        log_message_to_sentry('MVI Record Not Found', :warn, error_context(user))
+      when MVI::Errors::InvalidRequestError
```

Comment:
Are `MVI::Errors::InvalidRequestError` and `MVI::Errors::FailedRequestError` worthy of error level? An invalid request could be something that we need fix? Fully aware that I had it as warn level in the rails logger 😄

Reply:
I'm totally open to swapping these out, lets see the complaints roll in. :-p I fully anticipate replacing this with instrumentation in grafana vs sentry, but for now that extra content is super useful (MHV launch).